### PR TITLE
configury: add portals4 include path to osc-monitoring component build

### PR DESCRIPTION
**Pull request by tkordenbrock**
The osc-monitoring component didn't add the Portals4 include directory to AM_CPPFLAGS which causes a build failure if Portals4 is installed in a location that isn't in the compiler's default search path.

The PR clarifies the OPAL_CHECK_PORTALS4() prefix name, exports the Portals4 variables and adds the Portals4 CPPFLAGS to AM_CPPFLAGS.

Signed-off-by: Todd Kordenbrock <thkgcode@gmail.com>

@bosilca @jsquyres Please review.

### CODE REVIEW

File: ompi/mca/osc/monitoring/configure.m4 [hunk 1]
```diff
@@ -16,7 +16,11 @@ AC_DEFUN([MCA_ompi_osc_monitoring_CONFIG],[
...
           [$1],
           [$2])
-    OPAL_CHECK_PORTALS4([osc_monitoring],
+    OPAL_CHECK_PORTALS4([osc_monitoring_portals4],
                         [AC_DEFINE([OMPI_WITH_OSC_PORTALS4], [1], [Whether or not to generate template for osc_portals4])],
                         [])
+    # substitute in the things needed to build portals4
+    AC_SUBST([osc_monitoring_portals4_CPPFLAGS])
+    AC_SUBST([osc_monitoring_portals4_LDFLAGS])
+    AC_SUBST([osc_monitoring_portals4_LIBS])
```
Comment (jsquyres) [inline ompi/mca/osc/monitoring/configure.m4]: Doesn't look like the `LDFLAGS` and `LIBS` values are used.

**Comment by bosilca** [comment]
We do not need to interact with the other OSC modules. We are working on a fix.

**Comment by tkordenbrock** [comment]
@jsquyres PR updated to remove unused variables.

**Comment by tkordenbrock** [comment]
@bosilca So you are going to take out all the Portals4 references?

**Comment by bosilca** [comment]
@tkordenbrock yes all references to other OSC components.

**Comment by jsquyres** [comment]
@bosilca So should this PR be closed / replaced by a different PR?

**Comment by bosilca** [comment]
A more generic approach is under development in #4527 .

**Pull request closed without merging**